### DISCUSSION TITLE
CDT GCC Build Output Parser keeps clearing entries

### DISCUSSION THREAD

**Discussion by jozef-mitro** [eclipse-cdt/cdt#674] [Q&A]
When I use the CDT GCC Build Output Parser to get project defines it works just fine the first build after clean but during subsequent builds when a target is up to date, parser just ditches the previously found preprocessor entries. Any way to fix this?

Saw something about going to Project > Properties > Builders and setting Scanner Configuration Builder to be run only after a clean but that doesn't seem to do the job.

**Comment by jozef-mitro** [comment]
Update: Seems to be an issue in 2018-12 but works as expected in 2023-12. Must've been fixed/modified along the way.

**Comment by gvl2023** [comment]
I'm observing a similar behavior when using a Compilation Database Parser, which is seemingly dependent on the Build Output Parser. I see the issue with the 2023-12. The discovered entries disappear randomly, I haven't quite figured out the pattern that leads to it.
